docker/README.md

```diff
@@ -0,0 +1,73 @@
+# Docker
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+ubuntu:22.04
+
+## Build and run image
+
+### Build
+
+Building the base image is required.
+
+```
+docker build -t carbon-ubuntu2204-base ./ubuntu2204/base
+```
+
+Build image using git repository
+
+```bash
+docker build -t carbon ./ubuntu2204/github
+```
+
+Build image using copy instruction
+
+```bash
+docker build -f ./ubuntu2204/Dockerfile -t carbon ..
+```
+
+Run image
+
+```bash
+docker run carbon
+```
+
+Run image using specific file
+
+```bash
+docker run carbon bazel run //explorer -- ./explorer/testdata/print/format_only.carbon
+```
+
+### Using Volume
+
+```bash
+docker run -w /carbon-lang -v /path/to/carbon-lang:/carbon-lang carbon-ubuntu2004-base bazel run //explorer -- ./explorer/testdata/print/format_only.carbon
```

Comment:
It's a bit odd how you have this command, then all the others below. What do you think of doing something like:

```suggestion
# Run from the repository root for PWD.
cd ..
docker run -w /carbon-lang -v ${PWD}:/carbon-lang carbon-ubuntu2004-base bazel run //explorer -- ./explorer/testdata/print/format_only.carbon
```

That command should work for both bash and powershell. There's a bit of skew between `carbon` below and `carbon-ubuntu2004-base` here, but I feel like `base` is maybe a better choice since it's replacing /carbon-lang?

(note, I suggest also removing the commands below if you go this route, but github suggestions don't handle code blocks well)

Reply:
Yea I agree that they are a bit odd and should be cleand up, initially I left them for convenience.
I updated the commands and applied the suggestion.

The carbon example has the whole explorer already built during the build stages of the docker file while the base image does build the explorer only during runtime of the container.
For this example it might make more sense to just use the base image like you suggested (requires less steps also).